src/mam/mod_mam_rdbms_arch.erl

```diff
@@ -166,25 +167,70 @@ archive_message(Result, Host, MessID, UserID,
             {error, Reason}
     end.
 
-do_archive_message(_Result, Host, MessID, UserID, LocJID, RemJID, SrcJID, Dir, Packet) ->
-    Row = prepare_message(Host, MessID, UserID, LocJID, RemJID, SrcJID, Dir, Packet),
+do_archive_message(_Result, Host, MessID, UserID, LocJID, RemJID, SrcJID, OriginID, Dir, Packet) ->
+    Row = prepare_message(Host, MessID, UserID, LocJID, RemJID, SrcJID, OriginID, Dir, Packet),
     {updated, 1} = mod_mam_utils:success_sql_execute(Host, insert_mam_message, Row),
+    retract_message(Host, MessID, UserID, LocJID, RemJID, SrcJID, OriginID, Dir, Packet).
+
+retract_message(Host, _MessID, UserID, LocJID, RemJID, _SrcJID, _OriginID, Dir, Packet) ->
```

Comment:
Hmmm... I don't like passing parameters just to ignore them 😕 I see that this is how the async pool writer works though, so I don't really have a suggestion on how to fix both 🤷‍♂️

Reply:
I also don't like it but... it's good news - when we change the arguments to a map (ad I am going to do this soon), we could just pass the map here.